splicemachine/features/feature_store.py

```diff
@@ -528,6 +538,91 @@ def create_feature_set(self, schema_name: str, table_name: str, primary_keys: Di
+                           desc: Optional[str] = None, features: Optional[List[Feature]] = None) -> FeatureSet:
+        """
+        Creates and returns a new version of an existing feature set. Use this method when you want to make changes
+        to a deployed feature set.
+
+        :param schema_name: The schema under which to create the feature set table
+        :param table_name: The table name for this feature set
+        :param primary_keys: The primary key column(s) of this feature set
+        :param desc: The (optional) description
+        :param features: An optional list of features. If provided, any non-existant Features will be created with the Feature Set
+        :Example:
+            .. code-block:: python
+
+                from splicemachine.features import FeatureType, Feature
+                f1 = Feature(
+                    name='my_first_feature',
+                    description='the first feature',
+                    feature_data_type='INT',
+                    feature_type=FeatureType.ordinal,
+                    tags=['good_feature','a new tag', 'ordinal'],
+                    attributes={'quality':'awesome'}
+                )
+                f2 = Feature(
+                    name='my_second_feature',
+                    description='the second feature',
+                    feature_data_type='FLOAT',
+                    feature_type=FeatureType.continuous,
+                    tags=['not_as_good_feature','a new tag'],
+                    attributes={'quality':'not as awesome'}
+                )
+                feats = [f1, f2]
+                feature_set = fs.update_feature_set(
+                    schema_name='splice',
+                    table_name='foo',
+                    primary_keys={'MOMENT_KEY':"INT"},
+                    desc='test fset',
+                    features=feats
+                )
+
+        :return: FeatureSet
+        """
+        # database stores object names in upper case
+        schema_name = schema_name.upper()
+        table_name = table_name.upper()
+
+        features = [f.__dict__ for f in features] if features else None
+        fset_dict = { "primary_keys": {pk: sql_to_datatype(primary_keys[pk]) for pk in primary_keys},
+                      "description": desc,
+                      "features": features}
+
+        print(f'Registering feature set {schema_name}.{table_name} in Feature Store')
+        if features:
+            print(f'Registering {len(features)} features for {schema_name}.{table_name} in the Feature Store')
+        r = make_request(self._FS_URL, f'{Endpoints.FEATURE_SETS}/{schema_name}.{table_name}', RequestType.PUT, self._auth, body=fset_dict)
+        return FeatureSet(**r)
+
+    def alter_feature_set(self, schema_name: str, table_name: str, primary_keys: Optional[Dict[str, str]] = None,
+                           desc: Optional[str] = None, version: Optional[Union[str, int]] = None) -> FeatureSet:
+        """
```

Comment:
docs need to be reflected for the version param. Is the default latest? Add a `:param version`